chia/types/spend_bundle.py

```diff
@@ -77,6 +80,18 @@ def not_ephemeral_additions(self):
 
         return result
 
+    def get_memos(self) -> Dict[bytes32, bytes]:
+        memos: Dict[bytes32, bytes] = {}
+        for coin_spend in self.coin_spends:
+            result = Program.from_bytes(bytes(coin_spend.puzzle_reveal)).run(
+                Program.from_bytes(bytes(coin_spend.solution)))
+            error, result_human = parse_sexp_to_conditions(result)
+            assert error is None
+            for cvp in result_human:
+                if ConditionOpcode(cvp.opcode) == ConditionOpcode.CREATE_COIN and len(cvp.vars) > 2:
+                    memos[bytes32(cvp.vars[0])] = cvp.vars[2]
```

Comment:
This should probably be saved by the coin ID of the new coin rather than the puzzle hash.  There may be multiple of the same puzzle hash created and this would overwrite previous memos that had been saved.

Reply:
yeah that is what I meant to do. Thanks for pointing out the issue!